app/Models/Campaign.php

```diff
@@ -110,6 +110,21 @@ public function scopeWhereClosed($query)
             });
     }
 
+    /**
+     * Scope a query to only include campaigns containing specified cause within
+     * "cause" field.
+     */
+    public function scopeWhereHasCause($query, $cause)
+    {
+        // Sanitize the input to prevent melicious regex pattern injection:
+        $sanitizedCause = preg_replace("/[^\w-]/", '', $cause);
```

Comment:
Doesn't Laravel have some helper for this? Quick check and I found this [`prepareForValidation()`](https://laravel.com/docs/6.x/validation#prepare-input-for-validation) method. But I also think there's some quick helper, I just can't remember well and my google-foo is failing me, haha 😅

Reply:
Hmm interesting! I'm not sure where we'd put the `prepareForValidation` method here? I googled around for input sanitization because I honestly did expect there to be an easy built-in method, but couldn't really find anything beyond folks saying to use `pre_replace`.
But maybe an interesting spin on this might be to validate the request if the `has_cause` filter is set using something like [not_regex](https://laravel.com/docs/6.x/validation#rule-not-regex)?

Reply:
Sorry, I edited my question after since I realized it wasn't super clear... more-so meant possibility of just using `Str::slug()` method like it does in `prepareForValidation()` to just convert it to a slug-ified string, which I suspect would get rid of any regexy-ness 🤔

Reply:
My suggestion may not be super robust though, and `preg_replace()` may just be better!

Reply:
I tried this in Tinker and seemed to work well:

```php
>>> Illuminate\Support\Str::slug('(^|,)addiction(,|$)');
=> "addiction"
```

Reply:
Ohhh gotcha, that's an interesting idea! I think it'll indeed essentially be running the same sanitization to filter out any special characters which is convenient. I guess my one reservation is that it may be less clear as to what we're doing? I dunno, I'm kinda into it -- I assume our cause list will always be the same slug-like format!

Reply:
Maybe to keep it clear, don't inline it and keep the variable named `$sanitizedCause`, plus this way if the sanitized cause ends up being empty after the sanitization, you could do a check on this variable and eschew the query if it turns out it was bad (and now empty) input? 🤔

Reply:
I gave it a go in https://github.com/DoSomething/rogue/pull/985/commits/4411570e342c27a42ca63509f5fa3a260b622e59! If the string is empty, the filter just won't yield any matches since we'll be searching for a comma-separated empty string! 😌